xds/internal/xdsclient/authority.go

```diff
@@ -117,14 +118,34 @@ func newAuthority(args authorityArgs) (*authority, error) {
 		ServerCfg:          *args.serverCfg,
 		UpdateHandler:      ret.handleResourceUpdate,
 		StreamErrorHandler: ret.newConnectionError,
+		OnSendHandler:      ret.transportAfterSendHandler,
 		Logger:             args.logger,
 	})
 	if err != nil {
 		return nil, fmt.Errorf("creating new transport to %q: %v", args.serverCfg, err)
 	}
 	ret.transport = tr
 	return ret, nil
+}
 
+// transportAfterSendHandler is called by the underlying transport when it sends a
+// resource req successfully. Timers are started for resources waiting for a response.
+func (a *authority) transportAfterSendHandler(u *transport.UpdateChannelInfo) {
+	a.resourcesMu.Lock()
+	defer a.resourcesMu.Unlock()
+	rType := a.resourceTypeGetter(u.URL)
+	rTypeMap := a.resources[rType]
```

Comment:
Nit: s/rTypeMap/resourceStates/
The latter is the variable name used whenever we read stuff out of this two-level map.